src/nl/hannahsten/texifyidea/util/files/PsiFile.kt

```diff
@@ -268,4 +265,50 @@ fun PsiFile.getBibtexRunConfigurations() = project
     .filter { it.mainFile == findRootFile().virtualFile }
     .flatMap { it.bibRunConfigs }
     .map { it.configuration }
-    .filterIsInstance<BibtexRunConfiguration>()
\ No newline at end of file
+    .filterIsInstance<BibtexRunConfiguration>()
+
+/**
+ * Gets the smallest extractable expression at the given offset. this should reaaaly be a LatexFile
+ */
+fun PsiFile.expressionAtOffset(offset: Int): PsiElement? {
+    val element = findElementAt(offset) ?: return null
+
+    return element.parents(true)
+        .firstOrNull { it.elementType == LatexTypes.NORMAL_TEXT_WORD || it is LatexNormalText || it is LatexParameter || it is LatexMathContent || it is LatexCommandWithParams }
+}
+
+// should the reciever just be `LatexFile`?
```

Comment:
You could check the filetype but I think it's fine as it is, it just might not find anything if it's not the right thing